types/mithril/index.d.ts

```diff
@@ -1,863 +1,268 @@
-	* beyond the applicable DOM attributes.
-	*
-	* @see m
-	*/
-	interface Attributes {
-		/**
-		* The class name(s) for this virtual element, as a space-separated list.
-		*/
-		className?: string;
+	export interface Mount {
+		/** Mounts a component to a DOM element, enabling it to autoredraw on user events. */
+		(element: Element, component: ComponentTypes<any, any>): void;
+		/** Unmounts a component from a DOM element. */
+		(element: Element, component: null): void;
+	}
 
-		/**
-		* The class name(s) for this virtual element, as a space-separated list.
-		*/
-		class?: string;
+	/** Creates an event handler which takes the value of the specified DOM element property and calls a function with it as the argument. */
+	export type WithAttr = (name: string, callback: (value: any) => any, thisArg?: any) => (e: { currentTarget: any, [p: string]: any }) => void;
 
-		/**
-		* A custom, low-level configuration in case this element needs special
-		* cleanup after removal from the tree.
-		*
-		* @see ElementConfig
-		*/
-		config?: ElementConfig;
-
-		/**
-		* A key to optionally associate with this element.
-		*/
-		key?: string | number;
+	/** Returns an object with key/value pairs parsed from a string of the form: ?a=1&b=2 */
+	export type ParseQueryString = (queryString: string) => { [p: string]: any };
 
-		/**
-		* Any other virtual element properties, including attributes and event
-		* handlers.
-		*/
-		[property: string]: any;
-	}
+	/** Turns the key/value pairs of an object into a string of the form: a=1&b=2 */
+	export type BuildQueryString = (values: { [p: string]: any }) => string;
 
-	/**
-	* The basis of a Mithril controller instance.
-	*/
-	interface Controller {
-		/**
-		* An optional handler to call when the associated virtual element is
-		* destroyed.
-		*
-		* @param evt An associated event.
-		*/
-		onunload?(evt: Event): any;
+	export interface RequestOptions<T> {
+		/** The HTTP method to use. */
+		method?: string;
+		/** The data to be interpolated into the URL and serialized into the querystring (for GET requests) or body (for other types of requests). */
+		data?: any;
+		/** Whether the request should be asynchronous. Defaults to true. */
+		async?: boolean;
+		/** A username for HTTP authorization. */
+		user?: string;
+		/** A password for HTTP authorization. */
+		password?: string;
+		/** Whether to send cookies to 3rd party domains. */
+		withCredentials?: boolean;
+		/** Exposes the underlying XMLHttpRequest object for low-level configuration. */
+		config?(xhr: XMLHttpRequest): any;
+		/** Headers to append to the request before sending it. */
+		headers?: any;
+		/** A constructor to be applied to each object in the response. */
+		type?: new (o: any) => any;
+		/** A serialization method to be applied to data. Defaults to JSON.stringify, or if options.data is an instance of FormData, defaults to the identity function. */
+		serialize?(data: any): any;
```

Comment:
The return type should be whatever  can be sent through `XMLHttpRequest`, not `any`. (You can look it up in the [builtin `.d.ts` files](https://github.com/Microsoft/TypeScript/tree/master/lib).)